### DISCUSSION TITLE
Sws with axum ?

### DISCUSSION THREAD

**Discussion by FlorentDardenne** [static-web-server/static-web-server#302] [Q&A]
Hi,

I was more a web frontend than a backend dev. Familiar with fullstack Lamp setting for semi-pro usages. Since a couple of weeks, i’m looking around Rust.

My goal is to recreate a lamp server on a small vps for my small sites with Rust (personnals projects). I read some docs from SWS which speak to me : VirtualHost, Brotli, etc.…

Can we use SWS as a default web server ? And can we plug Axum web framework and database ?

Regards Florent,
( hope my english is understandable :o)

**Comment by joseluisq** [comment]
> Can we use SWS as a default web server ? And can we plug Axum web framework and database ?

SWS is a web server application for serving static files. In fact, it was created as an application (binary) project and never intended to be a library per se.
Although SWS is also published as a crate (library) to facilitate embedding, it is not suitable if you want to have control and flexibility like adding your own stuff on top of it as stated in the crate [docs](https://docs.rs/static-web-server/latest/static_web_server/#static-web-server-sws).

However, you could reuse several features from SWS library API like virtual host, static files, compression, etc and plug-in them to your own server in Axum omitting SWS server functionality which is highly coupled.

In the future, we could improve SWS to be a proper extensible library, for example introducing a middleware system to allow extensibility and reusability of functionality (features).
